private/pkg/manifest/manifest.go

```diff
@@ -76,6 +82,17 @@ func NewDigestFromString(typedDigest string) (*Digest, error) {
 	return NewDigestFromHex(hashfunc, digestStr)
 }
 
+func NewDigestFromBlobHash(hash *modulev1alpha1.Hash) (*Digest, error) {
```

Comment:
I'm thinking we should put this into its own file, perhaps `module.go`? manifest depends on storage and with this adds a dependency on our proto package. Separating out that dep may help in breaking a cycle later.

Also this function needs docs. (I should get [TCN-623](https://linear.app/bufbuild/issue/TCN-623/bufbuildbuf-is-not-linting-for-comments-on-exported-types) in flight.)

Reply:
> into its own file

You mean its own pkg? I had the same thought remembering your decision with `storage`. What about `private/pkg/manifest/manifestintegrations` for both `storage` and `proto`?

Reply:
Nah, just putting this kind of support into its own file in the manifest package. That way it's clear where dep lies and can be moved later if the dep needs to be broken.

Reply:
Done